vividus/src/main/java/org/vividus/transformer/AbstractFilteringTableTransformer.java

```diff
@@ -31,7 +34,13 @@
     protected List<String> filterColumnNames(List<String> allColumnNames, String byColumnNames)
     {
         List<String> filtered = new ArrayList<>(allColumnNames);
-        filtered.retainAll(Stream.of(StringUtils.split(byColumnNames, ';'))
+        String[] selectedColumnNames = StringUtils.split(byColumnNames, ';');
+
+        isTrue(allColumnNames.containsAll(Arrays.asList(selectedColumnNames)),
+                "The value of 'byColumnNames' - '%s' has one or more columns undefined in the example table",
+                byColumnNames);
+
+        filtered.retainAll(Stream.of(selectedColumnNames)
                 .map(String::trim)
                 .collect(Collectors.toList()));
         return filtered;
```

Comment:
can't we just return selectedColumnNames as list?

Reply:
> can't we just return selectedColumnNames as list?

I think the columns of the table should be returned in the original order, not the order that the users defined